### PR TITLE
create 2-3 new Documents for VFS,VGZ,SEA documents

### PR DESCRIPTION
**Pull request by walksanatora**
this is a draft so that i can recieve critiscism on how i wrote it, and get pointers for writing better documents (gonna finish them 1 by 1)
the first question is though, should i have VGZ be a diffrent document despite being VFS but GZipped

### CODE REVIEW

**Review by MCJack123 — CHANGES_REQUESTED** [review]
Right now, the document is very vague and doesn't really describe the data beyond "it has keys and values". There should be examples of what it looks like, covering any potential edge cases, as well as code to encode and decode it, including basic file reading/writing. It also needs some grammar improvements to match the professional tone requirements (mainly capitalization and punctuation).

1. VFS is fine to standardize, though we may want to standardize something more rich (including access/modification times, and possibly other possible metadata). I have a format that will be used in Phoenix (not published yet) which has metadata including permissions, but that may be out of scope for CraftOS-centric formats.
2. VGZ is unnecessary to standardize - it's just a gzipped VFS file. You may mention that gzip is a recommended compression format, but it's not a separate format itself.
3. As far as I know, SEA uses custom code to extract inside the file. We shouldn't be standardizing any specific code anywhere - we only provide protocols for coders to use so their different programs are able to communicate with each other. There's also no use case for programmers, as it only functions when executed, and only has a single purpose (extract files). In the end. it becomes a document saying that a file with extension `.sea` (which isn't even recognized as a program by default) *should* extract files, which isn't much of a standard.

**Comment by walksanatora** [comment]
updated document a bit
requesting recheck

**Comment by walksanatora** [comment]
1 possible change that i can make is simplifying my code into two seperate files so you dont have to look at my boilerplate to make sure arguments are correct

**Comment by walksanatora** [comment]
that and commenting my program

File: Standards/CCSMB-X.md [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+| MIME        | `appilcation/vfs`            |
+| Extensions  | `.vfs`                       |
+
+## Technical details
+
+The VFS file format is a text-based format that represents folder(s) as a lua table
+where the keys of the table are used as file names and values are used for conents.
+Directories in VFS are tables, whereas files have their contents embedded as a string.
+VFS files should be read and written in binary mode as to prevent converting binary charachters to `?`
+One drawback of the VFS format is that it does not save any MetaData of files compressed
```
Comment (MCJack123) [inline Standards/CCSMB-X.md:25]: I'd reword this to be a bit more positive:

> Note that the VFS format does not store metadata, so applications requiring metadata may need to use an alternative.

File: Standards/CCSMB-X.md [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+| Version     | 1.0.9                        |
+| Author      | walksanator (Walker Fowlkes) |
+| Type        | Virtual File System Archiv   |
+| MIME        | `appilcation/vfs`            |
+| Extensions  | `.vfs`                       |
+
+## Technical details
+
+The VFS file format is a text-based format that represents folder(s) as a lua table
+where the keys of the table are used as file names and values are used for conents.
```
Comment (MCJack123) [inline Standards/CCSMB-X.md:22]:
```suggestion
where the keys of the table are used as file names and values are used for contents.
```

File: Standards/CCSMB-X.md [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+| ----------- | -----------------------------|
+| Version     | 1.0.9                        |
+| Author      | walksanator (Walker Fowlkes) |
+| Type        | Virtual File System Archiv   |
+| MIME        | `appilcation/vfs`            |
+| Extensions  | `.vfs`                       |
+
+## Technical details
+
+The VFS file format is a text-based format that represents folder(s) as a lua table
```
Comment (MCJack123) [inline Standards/CCSMB-X.md:21]:
```suggestion
The VFS file format is a text-based format that represents folder(s) as a Lua table
```

File: Standards/CCSMB-X.md [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+where the keys of the table are used as file names and values are used for conents.
+Directories in VFS are tables, whereas files have their contents embedded as a string.
+VFS files should be read and written in binary mode as to prevent converting binary charachters to `?`
```
Comment (MCJack123) [inline Standards/CCSMB-X.md:24]:
```suggestion
VFS files should be read and written in binary mode as to prevent converting binary characters to `?`.
```

File: Standards/CCSMB-X.md [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+| Author      | walksanator (Walker Fowlkes) |
+| Type        | Virtual File System Archiv   |
+| MIME        | `appilcation/vfs`            |
+| Extensions  | `.vfs`                       |
+
+## Technical details
+
+The VFS file format is a text-based format that represents folder(s) as a lua table
+where the keys of the table are used as file names and values are used for conents.
+Directories in VFS are tables, whereas files have their contents embedded as a string.
```
Comment (MCJack123) [inline Standards/CCSMB-X.md:23]:
```suggestion
Directories in VFS are tables, whereas files have their contents embedded as strings.
```

File: Standards/CCSMB-X.md [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+VFS files should be read and written in binary mode as to prevent converting binary charachters to `?`
+One drawback of the VFS format is that it does not save any MetaData of files compressed
+
+## Examples included
```
Comment (MCJack123) [inline Standards/CCSMB-X.md:27]: I'd recommend having the examples embedded inline with descriptions of what they do. This makes it easier to read quickly.

**Review by MCJack123 — CHANGES_REQUESTED** [review]
Looking better, though it still has a few typos. Also, the examples are *way* too long - they should be less than 15 lines long, and show only one thing at a time. You should avoid machine-generated example files - write them out by hand with adequate spacing to make them readable.

Also, don't forget to mention recommending gzip for compression here.

-----
> requesting recheck

Heads up that this button exists: 
image

File: Standards/CCSMB-X.md (hunk 6 above)
Comment (walksanatora) [reply Standards/CCSMB-X.md:27]: but it is a full program, or should i have the Examples section contains example functions for creating/extracting (the gen_disk and unpack_vfs functions) along wit a small text example
and have the full example program and example image in the folder

File: Standards/CCSMB-X.md (hunk 5 above)
Comment (walksanatora) [reply Standards/CCSMB-X.md:23]: fixed

File: Standards/CCSMB-X.md [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+     
+*Last revised: 2023-01-DD*
+
+## Quick information
+
+| Information |                              |
+| ----------- | -----------------------------|
+| Version     | 1.0.9                        |
+| Author      | walksanator (Walker Fowlkes) |
+| Type        | Virtual File System Archiv   |
```
Comment (tomodachi94) [inline Standards/CCSMB-X.md:15]: Typo.
```suggestion
| Type        | Virtual File System Archive   |
```

**Review by tomodachi94 — CHANGES_REQUESTED** [review]

File: Standards/CCSMB-X.md (hunk 6 above)
Comment (MCJack123) [reply Standards/CCSMB-X.md:27]: You shouldn't be posting the full program here - put that on your own repo and link to it as an additional example. The examples here should be limited to single functions that demonstrate just one part at a time.

Comment (walksanatora) [reply Standards/CCSMB-X.md:27]: so should i embed the example functions i use to create/unpack the archive with comments
but *just* those functions
and include a simpler example since having a folder and 1 lua script and a 18kb text file is a bit mutch

Comment (MCJack123) [reply Standards/CCSMB-X.md:27]: Sure, but try to make them as simple and easy to read as possible (e.g. assume the file's correct, no error checking).

Comment (walksanatora) [reply Standards/CCSMB-X.md:27]: thats what they do allready
i do all the error checking outside of the functions anyway

Comment (walksanatora) [reply Standards/CCSMB-X.md]: i think it is better now

**Comment by Villain420** [comment]
I need a developer/programmer who can book multiple appointments to vfs global WhatsApp +8801822393145
